libs/sdk-ui-dashboard/src/presentation/widget/kpi/EditableDashboardKpi/EditableDashboardKpi.tsx

```diff
@@ -0,0 +1,183 @@
+    useWidgetExecutionsHandler,
+    selectEnableWidgetCustomHeight,
+    selectSeparators,
+    selectDisableKpiDashboardHeadlineUnderline,
+    useDashboardDispatch,
+    eagerRemoveSectionItem,
+    selectWidgetCoordinatesByRef,
+    selectFilterContextFilters,
+} from "../../../../model";
+import { DashboardItemHeadline, DashboardItemKpi } from "../../../presentationComponents";
+import { useDashboardComponentsContext } from "../../../dashboardContexts";
+
+import { useWidgetSelection } from "../../common/useWidgetSelection";
+import { ConfigurationBubble } from "../../common";
+import { KpiConfigurationPanel } from "./KpiConfigurationPanel/KpiConfigurationPanel";
+import { getKpiResult, getNoDataKpiResult, KpiRenderer, useKpiData } from "../common";
+import { IDashboardKpiProps } from "../types";
+
+export const EditableDashboardKpi = (props: IDashboardKpiProps) => {
+    const {
+        kpiWidget,
+
+        backend: customBackend,
+        workspace: customWorkspace,
+
+        ErrorComponent: CustomErrorComponent,
+        LoadingComponent: CustomLoadingComponent,
+
+        onError,
+    } = props;
+
+    const intl = useIntl();
+    const { ErrorComponent, LoadingComponent } = useDashboardComponentsContext({
+        ErrorComponent: CustomErrorComponent,
+        LoadingComponent: CustomLoadingComponent,
+    });
+
+    const backend = useBackendStrict(customBackend);
+    const workspace = useWorkspaceStrict(customWorkspace);
+
+    const dashboardFilters = useDashboardSelector(selectFilterContextFilters);
+
+    const {
+        error: kpiDataError,
+        result: kpiDataResult,
+        status: kpiDataStatus,
+    } = useKpiData({
+        kpiWidget,
+        backend,
+        dashboardFilters,
+        workspace,
+    });
+
+    const enableCompactSize = useDashboardSelector(selectEnableWidgetCustomHeight);
+    const separators = useDashboardSelector(selectSeparators);
+    const disableDrillUnderline = useDashboardSelector(selectDisableKpiDashboardHeadlineUnderline);
+    const isDrillable = kpiWidget.drills.length > 0;
+
+    const dispatch = useDashboardDispatch();
+    const coordinates = useDashboardSelector(selectWidgetCoordinatesByRef(widgetRef(kpiWidget)));
+    const onWidgetDelete = useCallback(() => {
+        dispatch(eagerRemoveSectionItem(coordinates.sectionIndex, coordinates.itemIndex));
+    }, [dispatch, coordinates.sectionIndex, coordinates.itemIndex]);
+
+    const { error, result, status } = useExecutionDataView(
+        {
+            backend,
+            workspace,
+            execution:
+                kpiDataStatus === "success"
+                    ? {
+                          seriesBy: compact([kpiDataResult!.primaryMeasure, kpiDataResult!.secondaryMeasure]),
+                          filters: kpiDataResult!.effectiveFilters,
+                      }
+                    : undefined,
+        },
+        [
+            kpiDataStatus,
+            kpiDataResult?.primaryMeasure,
```

Comment:
Maybe  you could spread primaryMeasure , secondaryMeasure and effectiveFilters before its added to dependency array.